### PR TITLE
feat: 🎸 get portfolio tx history

### PR DESCRIPTION
**Pull request by sansan**
### JIRA Link 

https://polymesh.atlassian.net/browse/DA-79

### Changelog / Description 

- adds endpoint to fetch the transaction history for a portfolio

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?
- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/portfolios/portfolios.controller.ts [hunk 1]
```diff
@@ -254,4 +255,41 @@ export class PortfoliosController {
...
+  })
+  @ApiOkResponse({
+    description: 'Portfolio transactions',
+    type: PortfolioModel,
+  })
+  @Get('/identities/:did/portfolios/:id/transactions')
+  async getTxHistory(
+    @Param() { did, id }: PortfolioDto,
+    @Query() { account, ticker }: GetTransactionsDto
+  ): Promise<ResultsModel<HistoricSettlement>> {
```
Comment (polymath-eric) [inline src/portfolios/portfolios.controller.ts:285]: There should be a `HistoricSettlementModel` that has the information with that. We shouldn't use the SDK return type as the controller return type.

**Comment by polymath-eric** [comment]
small comment but lgtm otherwise

File: src/portfolios/dto/get-transactions.dto.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  @IsOptional()
+  @IsString()
+  readonly account?: string;
+
+  @ApiPropertyOptional({
+    description: 'Asset ticker for which the transactions were made',
+    example: '123',
+  })
+  @IsOptional()
+  @IsString()
```
Comment (prashantasdeveloper) [inline src/portfolios/dto/get-transactions.dto.ts:20]:
```suggestion
  @IsTicker()
```

File: src/portfolios/portfolios.controller.ts (hunk 1 above)
Comment (prashantasdeveloper) [reply src/portfolios/portfolios.controller.ts:285]: Agreed, you will have to create a separate model class for this

File: src/portfolios/portfolios.controller.ts [hunk 3]
```diff
@@ -254,4 +255,41 @@ export class PortfoliosController {
...
+    description:
+      'The ID of the portfolio for which transactions are to be fetched. Use 0 for the default Portfolio',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiOkResponse({
+    description: 'Portfolio transactions',
+    type: PortfolioModel,
+  })
+  @Get('/identities/:did/portfolios/:id/transactions')
```
Comment (prashantasdeveloper) [inline src/portfolios/portfolios.controller.ts:281]: You missed adding the portfolio not found errors here

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

File: src/portfolios/models/historic-settlement.model.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  @ApiProperty({
+    description: 'Transaction status',
+    enum: SettlementResultEnum,
+    example: SettlementResultEnum.Executed,
+  })
+  readonly status: string;
+
+  @ApiProperty({
+    description: 'Transaction hash',
+    example: ['0x01'],
```
[the remark this answers is not in the document]
Comment (polymath-eric) [reply src/portfolios/models/historic-settlement.model.ts:35]: I guess the transaction hash and update the type probably makes more sense.

I am not sure how an account would get mapped to settlement clearly when its identity based

File: src/portfolios/dto/get-transactions.dto.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { IsOptional, IsString } from 'class-validator';
+
+import { IsTicker } from '~/common/decorators/validation';
+
+export class GetTransactionsDto {
+  @ApiPropertyOptional({
+    description: 'Account address involved in transactions',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
```
Comment (polymath-eric) [inline src/portfolios/dto/get-transactions.dto.ts:11]: Addresses should be ss58 encoded. `0x0600...` is a DID. Either the description or example needs to be updated.

File: src/portfolios/models/historic-settlement.model.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    description: 'Transaction status',
+    enum: SettlementResultEnum,
+    example: SettlementResultEnum.Executed,
+  })
+  readonly status: string;
+
+  @ApiProperty({
+    description: 'Array of account addresses involved in the settlement',
+    example: [
+      { address: '5grwXxxXxxXxxXxxXxxXxxXxxXxxXxxXxxXxxXxxXxxXxxXx' },
```
Comment (polymath-eric) [inline src/portfolios/models/historic-settlement.model.ts:36]: I like this as an example address. It looks good with the regular capital pattern.

**Review by polymath-eric — APPROVED** [review]

**Pull request merged**